src/app/CommandSender.cpp

```diff
@@ -77,20 +77,15 @@ CHIP_ERROR CommandSender::OnMessageReceived(Messaging::ExchangeContext * apExcha
     VerifyOrExit(aPayloadHeader.HasMessageType(Protocols::InteractionModel::MsgType::InvokeCommandResponse),
                  err = CHIP_ERROR_INVALID_MESSAGE_TYPE);
 
-    err = ProcessCommandMessage(std::move(aPayload), CommandRoleId::SenderId);
-    SuccessOrExit(err);
+    SuccessOrExit(err = ProcessCommandMessage(std::move(aPayload), CommandRoleId::SenderId));
 
 exit:
 
-    if (mpDelegate != nullptr)
+    if (mpCallback != nullptr)
     {
         if (err != CHIP_NO_ERROR)
         {
-            mpDelegate->CommandResponseError(this, err);
-        }
-        else
-        {
-            mpDelegate->CommandResponseProcessed(this);
+            mpCallback->OnError(this, Protocols::InteractionModel::Status::Failure, err);
```

Comment:
Do these changes mean we only get a callback on error and not success?  That seems like a problem.